src/mito2/src/region.rs

```diff
@@ -376,18 +399,92 @@ impl ManifestContext {
         }
 
         // Now we can update the manifest.
-        manager.update(action_list).await.inspect_err(
+        let version = manager.update(action_list).await.inspect_err(
             |e| error!(e; "Failed to update manifest, region_id: {}", manifest.metadata.region_id),
         )?;
 
-        if self.state.load() == RegionState::ReadOnly {
+        if self.state.load() == RegionRoleState::Follower {
             warn!(
-                "Region {} becomes read-only while updating manifest which may cause inconsistency",
+                "Region {} becomes follower while updating manifest which may cause inconsistency",
                 manifest.metadata.region_id
             );
         }
 
-        Ok(())
+        Ok(version)
+    }
+
+    /// Sets the [`RegionRole`].
+    ///
+    /// ```
+    ///     +------------------------------------------+    
+    ///     |                      +-----------------+ |    
+    ///     |                      |                 | |    
+    /// +---+------+       +-------+-----+        +--v-v---+
+    /// | Follower |       | Downgrading |        | Leader |
+    /// +---^-^----+       +-----+-^-----+        +--+-+---+
+    ///     | |                  | |                 | |    
+    ///     | +------------------+ +-----------------+ |    
+    ///     +------------------------------------------+    
+    ///
+    /// Transition:
+    /// - Follower -> Leader
+    /// - Downgrading Leader -> Leader
+    /// - Leader -> Follower
+    /// - Downgrading Leader -> Follower
+    /// - Leader -> Downgrading Leader
+    ///
+    /// ```
+    pub(crate) fn set_role_role_state(&self, next_role: RegionRole, region_id: RegionId) {
```

Comment:
Can we rename it to `set_role()` or `try_set_role()`? We only pass `RegionRole` to it.